lib/request.go

```diff
@@ -54,16 +54,7 @@ func (r *Request) SetBody(body io.Reader) {
 		rc = ioutil.NopCloser(body)
 	}
 	r.Body = rc
-	if body != nil {
-		switch v := body.(type) {
-		case *strings.Reader:
-			r.ContentLength = int64(v.Len())
-		case *bytes.Reader:
-			r.ContentLength = int64(v.Len())
-		case *bytes.Buffer:
-			r.ContentLength = int64(v.Len())
-		}
-	}
+	r.ContentLength = -1
```

Comment:
I think we discussed this error last week, in case that jogs your memory...  I observed that an error was being thrown by the REST API not being happy with the body length of some client requests.  I did some research and, apparently, a body length of -1 indicates that the body length is the length of the body.  Does that make sense?